drivers/watchdog/sunxi-twd.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <error.h>
+#include <mmio.h>
+#include <drivers/clock.h>
+#include <drivers/timer.h>
+#include <drivers/watchdog.h>
+#include <drivers/watchdog/sunxi-twd.h>
+
+#define CLOCK_CYCLES_MHZ    24
+
+#define TWD_STATUS_REG      0x00
+#define TWD_CTRL_REG        0x10
+#define TWD_RESTART_REG     0x14
+#define TWD_INTV_VAL_REG    0x30
+
+#define TWD_RESTART_KEY_VAL (0xD14 << 16)
+
+static void
+sunxi_twd_restart(void *param)
+{
+	uint32_t reg;
+
+	/* Cast param from periodic work function interface. */
+	struct device *dev = (struct device *)param;
+
+	/* Enable and perform restart. */
+	reg = TWD_RESTART_KEY_VAL | BIT(0);
+	mmio_write32(dev->regs + TWD_RESTART_REG, reg);
```

Comment:
There's got to be a way to make this fit on one line (`TWD_RESTART_KEY_VAL` → `TWD_RESTART_KEY`?)

Reply:
Not on _one line_, but _less lines_ if using `clearsetbits`.